openedx/features/journals/api.py

```diff
@@ -0,0 +1,276 @@
+        """
+        split_url = urlsplit(url)
+        override_url = urlunsplit((
+            split_url.scheme,
+            split_url.netloc,
+            JOURNALS_API_PATH,
+            split_url.query,
+            split_url.fragment,
+        ))
+        return override_url
+
+    def get_journals(self, orgs):
+        """
+        get_journals from discovery, filter on orgs is supplied
+        """
+        try:
+            if orgs:
+                response = self.client.journals.get(orgs=','.join(orgs), status='active')
+            else:
+                response = self.client.journals.get(status='active')
+            LOGGER.debug('response is type=%s', type(response))
+            return response.get('results')
+        except (HttpClientError, HttpServerError) as err:
+            LOGGER.exception(
+                'Failed to get journals from discovery-service [%s]',
+                err.content
+            )
+            return []
+
+    def get_journal_bundles(self, uuid=''):
+        """
+        get_journal_bundles from discovery on the base of uuid (optional)
+        """
+        try:
+            response = self.client.journal_bundles(uuid).get()
+        except (HttpClientError, HttpServerError) as err:
+            LOGGER.exception(
+                'Failed to get journal bundles from discovery-service [%s]',
+                err.content
+            )
+            return []
+        return response if uuid else response.get('results')
+
+
+class JournalsApiClient(object):
+    """
+    Class for interacting with the Journals Service
+    """
+    def __init__(self):
+        """
+        Initialize an authenticated Journals service API client by using the
+        provided user.
+        """
+        self.user = self.get_journals_worker()
+        jwt = JwtBuilder(self.user).build_token(['email', 'profile'], 16000)
+        self.client = EdxRestApiClient(
+            configuration_helpers.get_value('JOURNALS_API_URL', settings.JOURNALS_API_URL),
+            jwt=jwt
+        )
+
+    def get_journals_worker(self):
+        """ Return journals worker """
+        return User.objects.get(username=JOURNAL_WORKER_USERNAME)
+
+
+def fetch_journal_access(site, user):   # pylint: disable=unused-argument
+    """
+    Retrieve journal access record for given user.
+    Retrieve if from the cache if present, otherwise send GET request to the journal access api
+        and store it in the cache
+
+    Args:
+        site (Site)
+        user (username | str): user to retrieve access records for
+
+    Returns:
+        list of dicts: list of journal access dicts
+
+    Raises:
+        ConnectionError: raised if lms is unable to connect to the journals service.
+        SlumberBaseException: raised if API response contains http error status like 4xx, 5xx etc...
+        Timeout: raised if API is talking to long to respond
+    """
+    # TODO: WL-1560: (see jira for more info)
```

Comment:
Can you please put a very brief summary in this comment for posterity.

Reply:
fixed